processing/src/main/java/io/druid/query/groupby/epinephelinae/RowBasedGrouperHelper.java

```diff
@@ -93,7 +80,11 @@
         query.getDimensions().size(),
         querySpecificConfig.getMaxMergingDictionarySize() / (concurrencyHint == -1 ? 1 : concurrencyHint)
     );
-    final RowBasedColumnSelectorFactory columnSelectorFactory = new RowBasedColumnSelectorFactory();
+    final ThreadLocal<Row> columnSelectorRow = new ThreadLocal<>();
```

Comment:
DimensionSelectors made here may be used by multiple different processing threads, so the ThreadLocal is useful.